main.go

```diff
@@ -1,374 +1,357 @@
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	w.Write(raw)
+}
+
+func (p *Plugin) getPVTopology(PVName string) string {
+	return fmt.Sprintf("%s;<persistent_volume>", PVName)
+}
+
+func (p *Plugin) makeReport() (*report, error) {
+	go p.updateIops()
+	go p.updateLatency()
+	go p.updateThroughput()
+
+	metrics := make(map[string][]float64)
+	resource := make(map[string]node)
+
+	for Iops, iopsMetrics := range p.Iops {
+		metrics[p.getPVTopology(Iops)] = append(metrics[p.getPVTopology(Iops)], iopsMetrics.ReadIops, iopsMetrics.writeIops)
+	}
+
+	for Latency, latencyMetrics := range p.Latency {
+		metrics[p.getPVTopology(Latency)] = append(metrics[p.getPVTopology(Latency)], latencyMetrics.ReadLatency, latencyMetrics.WriteLatency)
+	}
+
+	for Throughput, throughputMetrics := range p.Throughput {
+		metrics[p.getPVTopology(Throughput)] = append(metrics[p.getPVTopology(Throughput)], throughputMetrics.ReadThroughput, throughputMetrics.WriteThroughput)
+	}
+
+	for pvName, _ := range metrics {
+		resource[pvName] = node{
+			Metrics: p.metrics(metrics[pvName]),
+		}
+	}
+	rpt := &report{
+		PersistentVolume: topology{
+			Nodes:           resource,
+			MetricTemplates: p.metricTemplates(),
+		},
+		Plugins: []pluginSpec{
+			{
+				ID:          "openebs",
+				Label:       "OpenEBS Plugin",
+				Description: "Adds graphs of metrics of OpenEBS PV",
+				Interfaces:  []string{"reporter"},
+				APIVersion:  "1",
+			},
+		},
+	}
+	return rpt, nil
+}
+
+// Create the Metrics type on top-left side
+func (p *Plugin) metrics(data []float64) map[string]metric {
+	metrics := map[string]metric{
+		"readIops": {
+			Samples: []sample{
+				{
+					Date:  time.Now(),
+					Value: data[0],
+				},
+			},
+			Min: 0,
+			Max: 100,
+		},
+		"writeIops": {
+			Samples: []sample{
+				{
+					Date:  time.Now(),
+					Value: data[1],
+				},
+			},
+			Min: 0,
+			Max: 100,
+		},
+		"readLatency": {
+			Samples: []sample{
+				{
+					Date:  time.Now(),
+					Value: data[2],
+				},
+			},
+			Min: 0,
+			Max: 100,
+		},
+		"writeLatency": {
+			Samples: []sample{
+				{
+					Date:  time.Now(),
+					Value: data[3],
+				},
+			},
+			Min: 0,
+			Max: 100,
+		},
+		"readThroughput": {
+			Samples: []sample{
+				{
+					Date:  time.Now(),
+					Value: data[4],
+				},
+			},
+			Min: 0,
+			Max: 100,
+		},
+		"writeThroughput": {
+			Samples: []sample{
+				{
+					Date:  time.Now(),
+					Value: data[5],
+				},
+			},
+			Min: 0,
+			Max: 100,
+		},
+	}
+	return metrics
+}
+
+func (p *Plugin) metricTemplates() map[string]metricTemplate {
+	return map[string]metricTemplate{
+		"readIops": {
+			ID:       "readIops",
+			Label:    "Iops(R)",
+			Format:   "",
+			Priority: 0.1,
+		},
+		"writeIops": {
+			ID:       "writeIops",
+			Label:    "Iops(W)",
+			Format:   "",
+			Priority: 0.2,
+		},
+		"readLatency": {
+			ID:       "readLatency",
+			Label:    "Latency(R)",
+			Format:   "",

```

Comment:
Why format is empty for Latency and throughput? 
I remember, you raised a PR for adding the format in Scope.